### PR TITLE
refactoring test: added function for extracting ids

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: src/gtfs/read.rs [hunk 1]
```diff
@@ -1416,4 +1378,10 @@ mod tests {
             );
         });
     }
+
+    fn extract_ids<T: Id<T>>(c: &Collection<T>) -> Vec<String> {
+        let mut ids: Vec<String> = c.values().map(|l| l.id().to_string()).collect();
+        ids.sort();
+        ids
+    }
```
Comment (datanel) [inline src/gtfs/read.rs]: @TeXitoi 
with this function I cannot get line ids from routes
I can do 
```rust
fn extract_ids<T: Id<T> + Id<U>, U: Id<U>>(c: &Collection<T>) -> Vec<String> {
    let mut ids: Vec<String> = c.values()
        .map(|l| <T as Id<U>>::id(l).to_string())
        .collect();
    ids.sort();
    ids
}

extract_ids::<Route, Line>(&collections.routes)
```
What do you think?

File: src/gtfs/read.rs [hunk 2]
```diff
@@ -1416,4 +1353,14 @@ mod tests {
             );
         });
     }
+
+    fn extract<'a, T, S: ::std::cmp::Ord + 'a>(f: fn(&'a T) -> S, c: &'a Collection<T>) -> Vec<S> {
+        let mut ids: Vec<S> = c.values().map(|l| f(l)).collect();
+        ids.sort();
+        ids
```
Comment (TeXitoi) [inline src/gtfs/read.rs]: Maybe `ids` is not a so good name.

File: src/gtfs/read.rs [hunk 3]
```diff
@@ -1416,4 +1353,14 @@ mod tests {
...
         });
     }
+
+    fn extract<'a, T, S: ::std::cmp::Ord + 'a>(f: fn(&'a T) -> S, c: &'a Collection<T>) -> Vec<S> {
+        let mut extracted_props: Vec<S> = c.values().map(|l| f(l)).collect();
+        extracted_props.sort();
+        extracted_props
+    }
+
+    fn extract_ids<'a, T: Id<T>>(c: &'a Collection<T>) -> Vec<&'a str> {
```
Comment (TeXitoi) [inline src/gtfs/read.rs]: I think lifetime can be elided.

File: src/gtfs/read.rs [hunk 4]
```diff
@@ -988,13 +988,10 @@ mod tests {
...
 
-            let mut commercial_modes: Vec<String> = collections
-                .commercial_modes
-                .values()
-                .map(|cm| cm.name.clone())
-                .collect();
-            commercial_modes.sort();
-            assert_eq!(commercial_modes, &["Bus", "Rail"]);
+            assert_eq!(
+                extract(|r| &r.name, &collections.commercial_modes),
```
Comment (TeXitoi) [inline src/gtfs/read.rs]: badly named arg `r`

File: src/gtfs/read.rs [hunk 5]
```diff
@@ -1006,22 +1003,15 @@ mod tests {
...
             assert_eq!(2, collections.physical_modes.len());
-            let mut physical_modes: Vec<String> = collections
-                .physical_modes
-                .values()
-                .map(|pm| pm.name.clone())
-                .collect();
-            physical_modes.sort();
-            assert_eq!(physical_modes, &["Bus", "Train"]);
+            assert_eq!(
+                extract(|r| &r.name, &collections.physical_modes),
```
Comment (TeXitoi) [inline src/gtfs/read.rs]: badly named arg `r`

File: src/gtfs/read.rs [hunk 6]
```diff
@@ -1416,4 +1353,14 @@ mod tests {
             );
         });
     }
+
+    fn extract<'a, T, S: ::std::cmp::Ord + 'a>(f: fn(&'a T) -> S, c: &'a Collection<T>) -> Vec<S> {
+        let mut extracted_props: Vec<S> = c.values().map(|l| f(l)).collect();
```
Comment (TeXitoi) [inline src/gtfs/read.rs]:
```rust
.map(f)
```

Comment (datanel) [reply src/gtfs/read.rs]:
```
error[E0619]: the type of this value must be known in this context
    --> src/gtfs/read.rs:1359:9
     |
1359 |         extracted_props.sort();
     |      
```

**Pull request merged**